substrate/windows/substrate-builder

```diff
@@ -0,0 +1,342 @@
+#!/usr/bin/env bash
+
+# Remove the defined logout command if possible to prevent
+# clearing the terminal when complete
+if [ -f /etc/bash.bash_logout ]; then
+    # NOTE: Don't exit on error here since it's only
+    # an attempt and failure does not affect the
+    # actual build
+    rm -f /etc/bash.bash_logout
+fi
+
+# Store substrates here
+output_dir="${1?Output directory is required}"
+# Find launchers here
+launcher_dir="${2?Launcher directory is required}"
+
+# Check for required tools and install if needed
+tools=( "git" "zip" "unzip" "python3" "curl" )
+needed=()
+
+for t in "${tools[@]}"; do
+    if ! command -v "${t}" > /dev/null; then
+        needed+=( "${t}" )
+    fi
+done
+
+if ! command -v dirname > /dev/null; then
+    needed+=( "coreutils" )
+fi
+
+if [ "${#needed[@]}" -gt 0 ]; then
+    pacman -S --noconfirm "${needed[@]}" || exit
+fi
+
+# Determine the root directory of the repository
+csource="${BASH_SOURCE[0]}"
+while [ -h "$csource" ] ; do csource="$(readlink "$csource")"; done
+root="$( cd -P "$( dirname "$csource" )/../../" && pwd )"
+
+printf "formatting paths to unix style...\n"
+# Clean the path if it's needed
+output_dir="$(cygpath -u "${output_dir}")"
+launcher_dir="$(cygpath -u "${launcher_dir}")"
+
+# Validate the launcher binaries exist before any
+# work is started
+if [ ! -f "${launcher_dir}/launcher-windows_386.exe" ] || [ ! -f "${launcher_dir}/launcher-windows_x86_64.exe" ]; then
+    printf "missing required vagrant launcher binaries in launcher directory (%s)" \
+        "${launcher_dir}" >&2
+    exit 1
+fi
+
+printf "building substrates and storing to: %s\n" "${output_dir}"
+
+# Create output directory if required and get full path
+if [ ! -d "${output_dir}" ]; then
+    mkdir -p "${output_dir}" || exit
+fi
+pushd "${output_dir}" > /dev/null || exit
+output_dir="$(pwd)" || exit
+popd > /dev/null || exit
+
+# Do work within an isolated work directory
+work_dir="$(mktemp -d vagrant-substrate.XXXXX)" || exit
+pushd "${work_dir}" > /dev/null || exit
+work_dir="$(pwd)" || exit
+
+# Grab styrene as it's what's building our substrate
+printf "checking out styrene...\n"
+git clone https://github.com/chrisroberts/styrene ./styrene  || exit
```

Comment:
is this fork still required? i can't tell if there's any difference between this and upstream